packages/near-membrane-base/src/membrane.ts

```diff
@@ -4212,82 +4233,15 @@ export function createMembraneMarshall(
-                                  // then `value` is not likely a prototype object.
-                                  return true;
-                              }
-                              // eslint-disable-next-line no-empty
-                          } catch {}
-                          try {
-                              if (ObjectHasOwn(target, LOCKER_LIVE_VALUE_MARKER_SYMBOL!)) {
-                                  return true;
-                              }
-                              // eslint-disable-next-line no-empty
-                          } catch {}
-                          try {
-                              if (
-                                  ReflectGetPrototypeOf(target) === null &&
-                                  // Ensure `value` is not an `Object.prototype` from an iframe.
-                                  (typeof constructor !== 'function' ||
-                                      constructor.prototype !== target)
-                              ) {
-                                  return true;
-                              }
-                              // eslint-disable-next-line no-empty
-                          } catch {}
-                          // We only check for regexp and array buffers here
-                          // since plain arrays and array buffer views are
-                          // marked as live in the BoundaryProxyHandler
-                          // constructor.
-                          //
-                          // Section 25.1.5.1 get ArrayBuffer.prototype.byteLength
-                          // https://tc39.es/ecma262/#sec-get-regexp.prototype.source
-                          // Step 3: If R does not have an [[OriginalSource]] internal slot, then
-                          //     a. If SameValue(R, %RegExp.prototype%) is true, return "(?:)".
-                          //     b. Otherwise, throw a TypeError exception.
-                          try {
-                              if (ObjectHasOwn(target, 'lastIndex')) {
-                                  ReflectApply(RegExpProtoSourceGetter, target, []);
-                                  return true;
-                              }
-                              // eslint-disable-next-line no-empty
-                          } catch {}
-                          // Section 25.1.5.1 get ArrayBuffer.prototype.byteLength
-                          // https://tc39.es/ecma262/#sec-get-arraybuffer.prototype.bytelength
-                          // Step 2: Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
-                          try {
-                              if ('byteLength' in target) {
-                                  ReflectApply(ArrayBufferProtoByteLengthGetter!, target, []);
-                                  return true;
-                              }
-                              // eslint-disable-next-line no-empty
-                          } catch {}
-                      }
```

Comment:
☝️ All of that goes is done in Locker anyways so now is handled by it. This reduces duplicate code and gives Locker the power to customize this check.